src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { formatString } from '../servises/formatString';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: FC<Props> = ({
+  people,
+  onSelected,
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useMemo(() => {
+    return debounce(setAppliedQuery, delay);
+  }, [delay]);
+
+  useEffect(() => {
+    return () => applyQuery.cancel();
+  }, [applyQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(() => {
+    if (appliedQuery.trim().length === 0) {
+      return people;
+    }
```

Comment:
Ensure that the filter is not called if the user enters spaces only. This is a requirement from the checklist: 'make sure that filter won't be called if user entered spaces only'.